apis/externalsecrets/v1alpha1/externalsecret_types.go

```diff
@@ -179,6 +179,7 @@ type ExternalSecretSpec struct {
 	DataFrom []ExternalSecretDataRemoteRef `json:"dataFrom,omitempty"`
 }
 
+// +kubebuilder:validation:Enum=Ready;Deleted
```

Comment:
Hmm..., ExternalSecretConditionType is used in the status, and I don't think we need validation for the field since we don't accept user input there.